libs/simulation-project-utils/simulation-project-utils/src/lib/ui/create-project/form-steps/uniform-time-course-simulation.component.ts

```diff
@@ -44,10 +45,11 @@ export class UniformTimeCourseSimulationComponent implements IFormStepComponent,
   }
 
   public populateFormFromFormStepData(formStepData: FormStepData): void {
-    this.formGroup.controls.initialTime.setValue(formStepData.initialTime);
-    this.formGroup.controls.outputStartTime.setValue(formStepData.outputStartTime);
-    this.formGroup.controls.outputEndTime.setValue(formStepData.outputEndTime);
-    this.formGroup.controls.numberOfSteps.setValue(formStepData.numberOfSteps);
+    // This method is called onInit and thus should set the default vals
```

Comment:
why is this needed if the default values are set in the contructor?  will this play nicely if the user wants to override the defaults?

Reply:
Great question. In angular forms, there is a difference between setting a default value (that which is specified in `constructor`) and setting a form value such that it actually renders. The default value specified in this component's constructor is only collected `onSubmit()` (called by the parent component), and thus will not render on the screen as a `Change` in the form data. This call to `.setValue()` originally set the data from the `formStepData` of the previous step, which was `null`. The form data source is actively subscribed to user changes, so this explicit setting will not mess with new override specifications.

TLDR; Explicitly setting these values is required for rendering the default values and will not tamper with user specifications.